bft/bft_interp/src/lib.rs

```diff
@@ -32,6 +32,17 @@ impl<T> Machine<T> {
     pub fn may_grow(&self) -> bool {
         self.may_grow
     }
+
+    /// Checks whether a provided program is valid Brainfuck code
+    ///
+    /// As it's a very thin wrapper around DecoratedProgram::from_program,
+    /// see that for examples of how this is used.
+    pub fn validate(&self, prog: &Program) -> Result<(), ParseError> {
+        // Reuse existing parsing logic, throwing away the success
+        // and propagating any errors
+        DecoratedProgram::from_program(prog)?;
+        Ok(())
+    }
```

Comment:
I'll admit I wasn't expecting to see this at all here.

Given the implication of a `DecoratedProgram` I would have expected to see the conversion in your main code, and then a change to the interpreter VM to take one of those directly.